kqueen/auth/ldap.py

```diff
@@ -0,0 +1,75 @@
+from .base import BaseAuth
+
+import ldap
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class LDAPAuth(BaseAuth):
+    def __init__(self, *args, **kwargs):
+        """
+        Implementation of :func:`~kqueen.auth.base.__init__`
+        """
+
+        super(LDAPAuth, self).__init__(*args, **kwargs)
+
+        if not hasattr(self, 'uri'):
+            raise Exception('Parameter uri is required')
+
+        self.connection = ldap.initialize(self.uri)
+
+    @staticmethod
+    def _email_to_dn(email):
+        """This function reads email and converts it to LDAP dn
+
+        Args:
+            email (str): e-mail address
+
+        Returns:
+            dn (str): LDAP dn, like 'cn=admin,dc=example,dc=org
+        """
+
+        segments = []
+
+        if '@' in email:
+            cn, dcs = email.split('@')
+        else:
+            cn = email
+            dcs = ''
+
+        if cn:
+            segments.append('cn={}'.format(cn))
+
+        if '.' in dcs:
+            for s in dcs.split('.'):
+                segments.append('dc={}'.format(s))
+
+        return ','.join(segments)
+
+    def verify(self, user, password):
+        """Implementation of :func:`~kqueen.auth.base.__init__`
+
+        This function tries to bind LDAP and returns result
```

Comment:
Could you add mark about function result. For me the result of logical verify returns nothing and reason of error in case verification is not made. Why do we return tuple with the user as a first item?

Reply:
Return values are defined in base class (https://github.com/Mirantis/kqueen/blob/ldap/kqueen/auth/base.py#L22) - is this what you want?

I choosed to return tthe user as well as error mesage to be able to provide more information about process, for example User logged in but password should be change.

Another case can be updating User - like writing some attributes to User or modification of `last_login` field in the future.